util/client/client.go

```diff
@@ -1,75 +1,111 @@
 
-	corev1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
 	apiextensionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
-	clientset "k8s.io/client-go/kubernetes"
+	metricsclientset "k8s.io/metrics/pkg/client/clientset/versioned"
 
 	"github.com/longhorn/longhorn-manager/datastore"
-
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
 	lhclientset "github.com/longhorn/longhorn-manager/k8s/pkg/client/clientset/versioned"
 	lhinformers "github.com/longhorn/longhorn-manager/k8s/pkg/client/informers/externalversions"
+	"github.com/longhorn/longhorn-manager/types"
 )
 
-type Client struct {
-	clients.Clients
-	Datastore *datastore.DataStore
+type Clients struct {
+	wranglerClients.Clients
+	MetricsClient *metricsclientset.Clientset
+	Scheme        *runtime.Scheme
+	Namespace     string
+	Datastore     *datastore.DataStore
+	StopCh        <-chan struct{}
 }
 
-func NewClient(ctx context.Context, config *rest.Config, namespace string, needDataStore bool) (*Client, error) {
-	if err := schemes.Register(corev1.AddToScheme); err != nil {
-		return nil, err
+func NewClients(kubeconfigPath string, stopCh <-chan struct{}) (*Clients, error) {
+	namespace := os.Getenv(types.EnvPodNamespace)
+	if namespace == "" {
+		logrus.Warnf("Cannot detect pod namespace, environment variable %v is missing, "+
+			"using default namespace", types.EnvPodNamespace)
+		namespace = corev1.NamespaceDefault
 	}
 
-	clients, err := clients.NewFromConfig(config, nil)
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "unable to get client config")
 	}
 
-	var ds *datastore.DataStore
+	config.Burst = 100
+	config.QPS = 50
 
-	if needDataStore {
-		kubeClient, err := clientset.NewForConfig(config)
-		if err != nil {
-			return nil, errors.Wrap(err, "unable to get k8s client")
-		}
+	if err := wranglerSchemes.Register(corev1.AddToScheme); err != nil {
+		return nil, err
+	}
 
-		extensionsClient, err := apiextensionsclientset.NewForConfig(config)
-		if err != nil {
-			return nil, errors.Wrap(err, "unable to get k8s extension client")
-		}
+	// Create k8s client
+	clients, err := wranglerClients.NewFromConfig(config, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get k8s client")
+	}
 
-		lhClient, err := lhclientset.NewForConfig(config)
-		if err != nil {
-			return nil, errors.Wrap(err, "unable to get lh client")
-		}
+	// Create Longhorn client
+	lhClient, err := lhclientset.NewForConfig(config)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get clientset")
+	}
 
-		kubeInformerFactory := informers.NewSharedInformerFactory(kubeClient, time.Second*30)
-		lhInformerFactory := lhinformers.NewSharedInformerFactory(lhClient, time.Second*30)
+	scheme := runtime.NewScheme()
+	if err := longhorn.SchemeBuilder.AddToScheme(scheme); err != nil {
+		return nil, errors.Wrap(err, "unable to create scheme")
```

Comment:
One question
Is it ok that call `AddToScheme` twice here?
cc @innobead @shuo-wu

Reply:
I think is okay to call it twice
AddToScheme function should be idempotent